src/components/molecules/modal.tsx

```diff
@@ -27,10 +27,35 @@ export const Modal = ({ children, className, hideCloseButton, name }: ModalProps
 		isOpen: state.modals[name],
 		onClose: state.closeModal,
 	}));
+
+	const modalRef = useRef<HTMLDivElement | null>(null);
+
 	const wrapperClass = cn("fixed left-0 top-0 z-50 flex h-full w-full items-center justify-center");
 	const modalClasses = cn("w-500 rounded-2xl border border-gray-950 bg-white p-3.5 text-gray-1250", className);
 	const bgClass = cn("absolute left-0 top-0 -z-10 h-full w-full bg-black/70");
 
+	useEffect(() => {
+		if (isOpen && modalRef.current) {
+			const firstInput = modalRef.current.querySelector("input, textarea, select");
+			if (!firstInput) return;
+
+			(firstInput as HTMLElement).focus();
+		}
+
+		const handleKeyDown = (event: KeyboardEvent) => {
+			if (event.key === "Escape" && isOpen) {
+				onClose(name);
```

Comment:
I think we should `stopPropagation` after we closed the modal, we don't want close other modals or other elements, but only the current modal.

Reply:
added